### PR TITLE
Valid Test Failing

### PR DESCRIPTION
**Pull request by caldwellc1**
My valid test is failing where it expects a 6 and gets back a 0. I think it has to do with not sending the matrix since the function doesn't take any values, but I'm not sure how to get around that or what to put instead. Any ideas?
Also I added a throw since you can only do the trace of a square matrix. I'm not sure why it's not listed on the API though.

### CODE REVIEW

File: Javatrix.java [hunk 1]
```diff
@@ -477,5 +477,31 @@ public Javatrix times(double scalar)
...
+     * Gets the trace of a square matrix.
+     *
+     * @throws IllegalArgumentException if rows and
+     * columns are not the same.
+     * @return the trace of the matrix.
+     **/
+    public double trace() throws IllegalArgumentException
+    {
+        m = this.m;
+        n = this.n;
```
Comment (KaraBeason) [inline Javatrix.java]: should this be m = this.getRowDimension() for consistency?  Same for n and colDimension

File: Javatrix.java [hunk 2]
```diff
@@ -477,5 +477,31 @@ public Javatrix times(double scalar)
...
+        double ace = 0;
+        if (m != n)
+        {
+            String ex = "Rows and Columns must be same length.";
+            throw new IllegalArgumentException(ex);
+        }
+        matrix = new double[m][n];
```
Comment (KaraBeason) [inline Javatrix.java]: line 498 is why the function is returning 0.  You don't need to set matrix equal to a new double[][] here.  Unit test passes w/o this line

**Review by KaraBeason — CHANGES_REQUESTED** [review]

**Comment by caldwellc1** [comment]
@KaraBeason Thanks that worked.

**Comment by KaraBeason** [comment]
@caldwellc1 no problem!  Looks good to me!

**Review by KaraBeason — APPROVED** [review]

**Pull request merged**